server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Uri.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.common.model;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+/** Uri container to enforce type checks*/
+@Slf4j
+public class Uri {
+    public static final Uri EMPTY = new Uri("");
+    // URI should be always decoded.
+    private final String uri;
+
+    public Uri(String uri) {
```

Comment:
How many of these will be created during a single processing?

Also https://www.slf4j.org/apidocs/src-html/org/slf4j/LoggerFactory.html#line-445

Reply:
Yep. logger is not needed there.
Do you have better suggestion than a wrapper class?